### ISSUE TITLE
`flutter create` template doesn't work with keyboard focus

### ISSUE THREAD

**Issue by tvolkert** [flutter/flutter#109911]
## Steps to reproduce

1. Create a new Flutter project with `flutter create`
2. Run the project on desktop
3. Try to tab to the plus button and press spacebar to press it

## Expected behavior

You expect the out-of-the-box app to support basic keyboard interaction (tab to traverse focus, spacebar to activate buttons, etc.) -- and you expect the UI to represent which element has the focus.

## Actual behavior

The app doesn't do all of the above.  It seems like maybe some elements can gain focus, but they don't paint it properly.

@anderdobo

**Comment by gspencergoog** [comment]
This is because the counter app predates focus support.  I'll fix it up.

**Comment by gspencergoog** [comment]
@tvolkert This appears to work for me on the master channel on Linux and Windows 11.  Can you give me some more details? i.e. what does `flutter doctor -v` say on the platform where it didn't work for you?

**Comment by gspencergoog** [comment]
Actually, I'm just going to close this on the assumption that it's fixed.  Please reopen if I'm wrong.

**Comment by anderdobo** [comment]
@gspencergoog I tested on MacOS and Windows and:
1.  Focus traverses to the plus button as expected
2. Pressing the space bar activates the plus button as expected

However, the current UI does not actually show which UI element has focus even when one does. Attaching examples from Windows and MacOS where the UI does show which element has focus.

**Comment by gspencergoog** [comment]
@anderdobo Are you sure that it doesn't show when it has focus?  Here are screen shots of the current example UI on Windows 11, which does show a change in color when the focus is on the "+" button:

When the "+" doesn't have focus:
<img width="350" alt="CounterNoHighlight" src="https://user-images.githubusercontent.com/8867023/189768590-f495c000-050e-4287-88f5-d9e57241f4af.png">

When the "+" does have focus:
<img width="350" alt="CounterHighlight" src="https://user-images.githubusercontent.com/8867023/189768631-341a7fea-444a-449f-9808-22d1a0ded12a.png">

Note that the highlighted button is slightly darker.

It is a subtle difference for sure, even though it is the correct value from the Material spec, so below I set the `focusColor` to pink next to show that it really is highlighting it.

When the "+" has focus, and I've set the `focusHighlight` color to pink:
<img width="350" alt="CounterHighlightRed" src="https://user-images.githubusercontent.com/8867023/189768672-23682535-7b18-4fba-9be3-9a56b725357f.png">

**Comment by anderdobo** [comment]
Ah! You're right @gspencergoog . The color difference on focus is barely perceptible, and not apparent when tabbing further since it's the only focusable element. This is functioning as needed (color in this app aside).
